src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1882,16 +1882,25 @@ MavlinkReceiver::handle_message_heartbeat(mavlink_message_t *msg)
 		mavlink_msg_heartbeat_decode(msg, &hb);
 
 		/* ignore own heartbeats, accept only heartbeats from GCS */
-		if (msg->sysid != mavlink_system.sysid && hb.type == MAV_TYPE_GCS) {
+		if (msg->sysid != mavlink_system.sysid || hb.type == MAV_TYPE_ONBOARD_CONTROLLER) {
```

Comment:
Why did `MAV_TYPE_GCS` got removed? It means we'll accept heartbeats from other drones as telemetry status. I think we should limit to what we accept to our sysid or a GCS.
Something like:
```suggestion
		if ((msg->sysid != mavlink_system.sysid && hb.type == MAV_TYPE_GCS) || (msg->sysid == mavlink_system.sysid && hb.type == MAV_TYPE_ONBOARD_CONTROLLER)) {
```

Reply:
your suggestion make sense to me. 
Done here 982e31587d80aa03441aeb6d6f4e6b2ceadae89e